app/src/main/java/com/android/voyageur/ui/trip/schedule/WeeklyView.kt

```diff
@@ -93,107 +126,225 @@ private fun WeekCard(
+
+  Box(
+      modifier =
+          Modifier.width(130.dp)
+              .height(19.dp)
+              .background(
+                  color =
+                      ButtonDefaults.buttonColors()
+                          .containerColor
+                          .copy(alpha = if (isInTrip) 1f else 0.5f),
+                  shape = RoundedCornerShape(size = 25.dp))
+              .then(
+                  if (isInTrip) {
+                    Modifier.clickable {
+                      tripsViewModel.selectDay(currentDate)
+                      navigationActions.navigateTo(Screen.ACTIVITIES_FOR_ONE_DAY)
+                    }
+                  } else Modifier),
+      contentAlignment = Alignment.CenterStart) {
+        DayText(dayText, isInTrip)
+      }
+}
+
+/**
+ * Displays the formatted text for a day with appropriate styling based on trip status.
+ *
+ * @param dayText The formatted text to display for the day
+ * @param isInTrip Whether this day falls within the trip dates
+ */
+@Composable
+private fun DayText(dayText: String, isInTrip: Boolean) {
+  Text(
+      text = dayText,
+      style =
+          TextStyle(
+              fontSize = 10.sp,
+              lineHeight = 20.sp,
+              fontWeight = FontWeight(500),
+              color =
+                  MaterialTheme.colorScheme.inverseOnSurface.copy(
+                      alpha = if (isInTrip) 1f else 0.5f),
+              letterSpacing = 0.1.sp,
+          ),
+      maxLines = 1,
+      overflow = TextOverflow.Ellipsis,
+      modifier = Modifier.padding(horizontal = 10.dp))
+}
+
+/**
+ * Filters activities to find those occurring on a specific date.
+ *
+ * @param activities List of all activities to filter
+ * @param currentDate The date to filter activities for
+ * @return List of activities occurring on the specified date
+ */
+private fun getActivitiesForDay(
+    activities: List<Activity>,
+    currentDate: LocalDate
+): List<Activity> {
+  return activities.filter { activity ->
+    val activityDate =
+        activity.startTime.toDate().toInstant().atZone(ZoneId.systemDefault()).toLocalDate()
+    activityDate == currentDate
+  }
+}
+
+/**
+ * Builds the display text for a day in the weekly view. Format varies based on whether the day is
+ * within the trip dates and has activities.
+ *
+ * @param date The date to generate text for
+ * @param activityCount Number of activities on this date
+ * @param isInTrip Whether the date falls within the trip's date range
+ * @return Formatted string representation of the day (e.g., "M 15 - 2 activities" or "M 15")
+ */
+private fun buildDayText(date: LocalDate, activityCount: Int, isInTrip: Boolean): String {
+  val dayInitial = date.dayOfWeek.toString().take(1)
+  val dayNumber = date.dayOfMonth
+  return if (isInTrip && activityCount == 1) {
+    "$dayInitial $dayNumber - $activityCount activity"
+  } else if (isInTrip && activityCount > 1) {
+    "$dayInitial $dayNumber - $activityCount activity"
```

Comment:
Here the text should be x activities not activity, i.e. the plural form of the noun

Reply:
Thank you for noticing it!